src/hotspot/share/classfile/packageEntry.cpp

```diff
@@ -255,38 +270,38 @@ static int compare_package_by_name(PackageEntry* a, PackageEntry* b) {
 }
 
 void PackageEntryTable::iterate_symbols(MetaspaceClosure* closure) {
-  for (int i = 0; i < table_size(); ++i) {
-    for (PackageEntry* p = bucket(i); p != NULL; p = p->next()) {
+  auto syms = [&] (const Symbol*& key, PackageEntry*& p) {
       p->iterate_symbols(closure);
-    }
-  }
+  };
+  _table.iterate_all(syms);
 }
 
 Array<PackageEntry*>* PackageEntryTable::allocate_archived_entries() {
   // First count the packages in named modules
-  int n, i;
-  for (n = 0, i = 0; i < table_size(); ++i) {
-    for (PackageEntry* p = bucket(i); p != NULL; p = p->next()) {
-      if (p->module()->name() != NULL) {
-        n++;
-      }
+  int n = 0;
+  auto count = [&] (const Symbol*& key, PackageEntry*& p) {
+    if (p->module()->name() != NULL) {
```

Comment:
The better way to check for the unnamed module is to use the ModuleEntry::is_named() method.

Reply:
Fixed.